assignments/prototype-refactor.js

```diff
@@ -4,6 +4,179 @@ Prototype Refactor
 
+/*
+  === GameObject ===
+  * createdAt
+  * dimensions (These represent the character's size in the video game)
+  * destroy() // prototype method -> returns the string: 'Object was removed from the game.'
+  * 
+*/
+
+/*
+function GameObject(attrs) {
+    this.createdAt = attrs.createdAt;
+    this.dimensions = attrs.dimensions;
+  }
+  
+  GameObject.prototype.destroy = function() {
+    return "Object was removed from the game.";
+  };
+  /*
+    === CharacterStats ===
+    * healthPoints
+    * name
+    * takeDamage() // prototype method -> returns the string '<object name> took damage.'
+    * should inherit destroy() from GameObject's prototype
+  */
+  
+  /*
+  function CharacterStats(CharacterData) {
+    GameObject.call(this, CharacterData);
+  
+    this.healthPoints = CharacterData.healthPoints;
+    this.name = CharacterData.name;
+  }
+  
+  CharacterStats.prototype = Object.create(GameObject.prototype);
+  
+  CharacterStats.prototype.takeDamage = function() {
+    return this.name + " took damage."; 
+  };
+  
+  
+  /*
+    === Humanoid (Having an appearance or character resembling that of a human.) ===
+    * team
+    * weapons
+    * language
+    * greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
+    * should inherit destroy() from GameObject through CharacterStats
+    * should inherit takeDamage() from CharacterStats
+  */
+ /*
+   
+  function Humanoid(HumanoidStats) {
+    CharacterStats.call(this, HumanoidStats); //?
+    this.team = HumanoidStats.team;
+    this.weapons = HumanoidStats.weapons;
+    this.language = HumanoidStats.language; 
+  }
+  Humanoid.prototype = Object.create(CharacterStats.prototype);
+  
+  Humanoid.prototype.greet = function() {
+    return this.name + " offers a greeting in " + this.language; //? 
+  };
+*/
+  /*
+    * Inheritance chain: GameObject -> CharacterStats -> Humanoid
+    * Instances of Humanoid should have all of the same properties as CharacterStats and GameObject.
+    * Instances of CharacterStats should have all of the same properties as GameObject.
+  */
+  
+  
+  
+  // Test you work by un-commenting these 3 objects and the list of console logs below:
+  
+/*
+
 2. Your goal is to refactor all of this code to use ES6 Classes. The console.log() statements should still return what is expected of them.
 
 */
+
+class GameObject {
+    constructor(gameObjectProperties){
+        this.createdAt = gameObjectProperties.createdAt;
+        this.dimensions = gameObjectProperties.dimensions;
+    }
+
+    destroy() {
+        return "Object was removed from the game.";
+    }
+}
+
+class CharacterStats extends GameObject {
+    constructor(CharacterProperties) {
+        super(CharacterProperties);
+        this.healthPoints = CharacterProperties.healthPoints;
+        this.name = CharacterProperties.name; 
+    }
+
+    takeDamage() {
+        return this.name + " took damage."; 
```

Comment:
ES 6 challenge! Try to refactor into a string literal if you have time and would like random practice with ES6